src/al.jl

```diff
@@ -95,16 +95,16 @@ function al(nlp :: AbstractNLPModel; max_iter :: Int = 1000, max_time :: Real =
     end
 
     # stationarity measure
-    gL = grad(nlp, x) - jtprod(nlp, x, al_nlp.y)
-    project_step!(gp, x, -gL, T.(nlp.meta.lvar), T.(nlp.meta.uvar)) # Proj(x - gL) - x
+    gL = grad(nlp, al_nlp.x) - jtprod(nlp, al_nlp.x, al_nlp.y)
+    project_step!(gp, al_nlp.x, -gL, T.(nlp.meta.lvar), T.(nlp.meta.uvar)) # Proj(x - gL) - x
```

Comment:
It seems easier and clearer to leave `x`. Notice that atributing `x = S.solution`, for instance, doesn't cost much, because it simply changes the reference of `x`.

Reply:
I prefer to use al_nlp.x, because it already exists.